### PR TITLE
Cross-platform pygame viewer

### PR DESCRIPTION
**Pull request by martyn**
This introduces a pygame real-time sampling and training viewer.  It works on linux and osx, it should work on windows as well once hypergan supports it.  Remaining:

* [x] default viewer, refactor
* [x] variable window size
* [x] set title
* [x] update instructions
* [x] pip dependencies
* [ ] icon

### CODE REVIEW

**Comment by martyn** [comment]
I generated this on osx in cpu with :

```
  hypergan sample [dataset] --resize --size 256x256x3 --sampler batch -c face-0.9 -b 8 --sampler random_walk --save_samples
```

output

**Comment by martyn** [comment]
No icon yet but I don't want to hold up merging because of that.

File: hypergan/samplers/random_walk_sampler.py [hunk 1]
```diff
@@ -9,32 +9,38 @@ def __init__(self, gan, samples_per_row=8):
         self.y = None
         self.x = None
         self.step = 0
-        self.steps = 30
+        self.steps = 8
         self.target = None
 
     def _sample(self):
         gan = self.gan
-        z_t = gan.encoder.sample
```
Comment (mikkel) [inline hypergan/samplers/random_walk_sampler.py]: The difference between `uniform_encoder` and `encoder` is confusing, especially when we don't check that this is alphagan.

Comment (martyn) [inline hypergan/samplers/random_walk_sampler.py]: Yep I'm going to revert this file for now.  RandomWalk looks bad on AlphaGAN without switching encoders/samples.  Maybe we need to refactor samplers to work better on different architectures.

**Pull request merged**